java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -594,7 +585,9 @@ private void visit(Tree tree, @Nullable Tree terminator) {
         executeUnaryExpression(tree);
         break;
       case IDENTIFIER:
-        executeIdentifier((IdentifierTree) tree);
+        if(executeIdentifier((IdentifierTree) tree)) {
+          return;
```

Comment:
I don't like the fact that the method is now interrupting visit. I would appreciate at least a documentation of what means the returned boolean value (it's not at all explicit from the method name)